<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Building supermatrix matrices using the phylotools package • phylotools</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Building supermatrix matrices using the phylotools package">
<meta property="og:description" content="phylotools">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">phylotools</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.2.4</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/phylotools-vignette.html">Building supermatrix matrices using the phylotools package</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/helixcn/phylotools/">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="phylotools-vignette_files/header-attrs-2.7/header-attrs.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Building supermatrix matrices using the phylotools package</h1>
                        <h4 class="author">Jinlong Zhang</h4>
            
            <h4 class="date">2021-03-24</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/helixcn/phylotools/blob/master/vignettes/phylotools-vignette.Rmd"><code>vignettes/phylotools-vignette.Rmd</code></a></small>
      <div class="hidden name"><code>phylotools-vignette.Rmd</code></div>

    </div>

    
    
<p>Building phylogenetic trees usually require multiple genes and the supermatrix approach is one of the popular ways to incorporate the information. However, the preparation of supermatrix is not straightforward and error-prone. The phylotools package aims to simplify the preparation of supermatrix. It automatically concatenated different gene markers by their names, i.e., building supermatrices based on the aligned Fasta or Phylip files.</p>
<div id="the-relaxed-phylip-format" class="section level2">
<h2 class="hasAnchor">
<a href="#the-relaxed-phylip-format" class="anchor"></a>The Relaxed Phylip Format</h2>
<p>The resulted supermatrices files are in the Relaxed Phylip format, a format that could be used in RAxML and IQ-Tree. The Relaxed Phylip format is an extension of the strict Phylip format, i.e., the two numbers in the first line show the number of sequences and the number of base pairs (bp), respectively, with a white-space in between. The second line starts with the sequence name, and between the sequence name and the nucleotide sequences, there is a white-space. In the strict Phylip format, each sequence name only allows 10 characters, software such as Phylip (<a href="https://evolution.genetics.washington.edu/phylip.html" class="uri">https://evolution.genetics.washington.edu/phylip.html</a>) will truncate the name and only take the first 10 characters if there are more than 10. The Relaxed Phylip format does not have this limitation and the names can contain as long as 128 characters.</p>
</div>
<div id="why-building-supermatrix" class="section level2">
<h2 class="hasAnchor">
<a href="#why-building-supermatrix" class="anchor"></a>Why building supermatrix?</h2>
<p>Why is it necessary to use multiple gene markers to build an evolutionary tree? This is mainly because the individual genes markers are generally too short to fully reflect the relationships of the species, and building more accurate evolutionary trees generally requires longer sequences. Obtaining DNA barcoding markers became relatively easy from the year 2005 on wards, and these markers are suitable for building large phylogenies.</p>
</div>
<div id="commonly-used-dna-barcoding-markers" class="section level2">
<h2 class="hasAnchor">
<a href="#commonly-used-dna-barcoding-markers" class="anchor"></a>Commonly used DNA barcoding markers</h2>
<p>DNA barcoding sequences are generally obtained by the traditional Sanger sequencing method based on PCR products. rbcLa (Ribulose bisphosphate carboxylase large chain), matK (Megakaryocyte-Associated Tyrosine Kinase) and trnH-psbA (intergenic spacer region) are among the most commonly used DNA barcoding markers, and they all derived from the chloroplast genome. Other markers, such as ITS (Internal transcribed spacer), are also commonly used.</p>
</div>
<div id="the-gene-markers-mutation-rate" class="section level2">
<h2 class="hasAnchor">
<a href="#the-gene-markers-mutation-rate" class="anchor"></a>The gene marker’s mutation rate</h2>
<p>The mutation rates could vary greatly among different genes. Some genes are more stable and the differences between families and genera could be very minor, while others mutate rapidly and could be very different even among populations/individuals of the same species. For example, <strong>rbcLa</strong> is commonly used to locate the family a taxon belongs to, while <strong>matK</strong> could locate the genus or even species. Non-coding genes with very fast mutation rates, such as the <strong>trnH-psbA</strong>, are often used to determine species or subspecies. Using partitioned model during phylogenetic inferences is recommended.</p>
</div>
<div id="missing-data" class="section level2">
<h2 class="hasAnchor">
<a href="#missing-data" class="anchor"></a>Missing Data</h2>
<p>Ideally, all the markers of a species should be obtained. However, some sequences of some species may not always be amplified or sequenced successfully, resulting in missing data. These missing loci will be denoted as <code><a href="https://rdrr.io/r/utils/Question.html">?</a></code> in the supermatrix.</p>
</div>
<div id="similar-software" class="section level2">
<h2 class="hasAnchor">
<a href="#similar-software" class="anchor"></a>Similar software</h2>
<p>In addition to phylotools, the following software can also be used to build supermatrix.</p>
<ol style="list-style-type: decimal">
<li>
<em>Biopython</em> (<a href="https://biopython.org/wiki/Concatenate_nexus" class="uri">https://biopython.org/wiki/Concatenate_nexus</a>)</li>
<li>
<em>Utensils</em> (<a href="https://github.com/ballesterus/Utensils" class="uri">https://github.com/ballesterus/Utensils</a>)</li>
<li>
<em>Geneious</em> (<a href="https://assets.geneious.com/manual/11.1/GeneiousManualsu53.html" class="uri">https://assets.geneious.com/manual/11.1/GeneiousManualsu53.html</a>)</li>
<li>
<em>catfasta2phyml</em> (<a href="https://github.com/nylander/catfasta2phyml" class="uri">https://github.com/nylander/catfasta2phyml</a>)</li>
<li>
<em>SEDA</em> (<a href="http://www.sing-group.org/seda/" class="uri">http://www.sing-group.org/seda/</a> , <a href="https://www.biostars.org/p/332853/" class="uri">https://www.biostars.org/p/332853/</a>)</li>
<li>
<em>SuperMatrix</em>: (<a href="http://coleoguy.blogspot.com/2015/04/r-concatenate-alignments-into.html" class="uri">http://coleoguy.blogspot.com/2015/04/r-concatenate-alignments-into.html</a>)</li>
<li>
<em>concat</em> (<a href="https://figshare.com/articles/concatenation_R/3839538" class="uri">https://figshare.com/articles/concatenation_R/3839538</a>)</li>
<li>
<em>AMAS</em> (<a href="https://www.ncbi.nlm.nih.gov/pmc/articl" class="uri">https://www.ncbi.nlm.nih.gov/pmc/articl</a>::es/PMC4734057/)</li>
</ol>
<p>In which Genious has a graphical interface.</p>
</div>
<div id="features-of-phylotools" class="section level2">
<h2 class="hasAnchor">
<a href="#features-of-phylotools" class="anchor"></a>Features of Phylotools</h2>
<ul>
<li>
<code>phylotools</code> is written entirely in R and is easy to use.</li>
<li>
<code>phylotools</code> assumes that the same species has the same name in different fasta or phylip files.</li>
<li>
<code>phylotools</code> can recognize the aligned fasta or phylip files, and the generated supermatrix is saved in the Relaxed Phylip format.</li>
</ul>
</div>
<div id="the-installation" class="section level2">
<h2 class="hasAnchor">
<a href="#the-installation" class="anchor"></a>The Installation</h2>
<p>The latest version of <code>phylotools</code> is stored on github and has not been uploaded to CRAN. Use the following command to install.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">devtools</span><span class="fu">::</span><span class="fu"><a href="https://devtools.r-lib.org//reference/remote-reexports.html">install_github</a></span><span class="op">(</span><span class="st">"helixcn/phylotools"</span><span class="op">)</span></code></pre></div>
<p>If <code>devtools</code> has not been installed, you can use <code><a href="https://rdrr.io/r/utils/install.packages.html">install.packages("devtools")</a></code> to install.</p>
</div>
<div id="functions-in-phylotools" class="section level2">
<h2 class="hasAnchor">
<a href="#functions-in-phylotools" class="anchor"></a>Functions in phylotools</h2>
<ul>
<li>
<code>clean.fasta.name</code> removes special characters from fasta sequence names so that they can be entered and displayed properly when building trees</li>
<li>
<code>dat2fasta</code> converts DNA sequences data.frame to fasta files</li>
<li>
<code>dat2phylip</code> converts DNA sequences data.frame to phylip files</li>
<li>
<code>get.fasta.name</code> obtains the names of all sequences in the fasta file</li>
<li>
<code>get.phylip.name</code> obtains the names of all sequences in the phylip file</li>
<li>
<code>read.fasta</code> reads the fasta file</li>
<li>
<code>read.phylip</code> reads the phylip file</li>
<li>
<code>rename.fasta</code> renames the sequences within a fasta file according to a data frame supplied.</li>
<li>
<code>rm.sequence.fasta</code> removes a specific sequence from a fasta file</li>
<li>
<code>split_dat</code> splits the data into separate fasta files based on the grouping of the sequence, e.g., for trnH-psbA sequences grouped by Orders (a taxonomic rank).</li>
<li>
<code>sub.taxa.label</code> substitutes the taxa name in the phylogenetic tree (a <code>phylo</code> object of <code>ape</code>)</li>
<li>
<code>supermat</code> creates supermatrix and RAxML partition files from aligned fasta or phylip files.</li>
</ul>
</div>
<div id="creating-supermatrix-with-phylip-files" class="section level2">
<h2 class="hasAnchor">
<a href="#creating-supermatrix-with-phylip-files" class="anchor"></a>Creating supermatrix with phylip files</h2>
<p>Suppose there are three Phylip files encoding different genes, create the supermatrix of the three genes and save them in Relaxed Phylip format, and generate RAxML Partition File to record the start and end loci of each gene.</p>
<p>The corresponding codes are as follows.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/helixcn/phylotools">phylotools</a></span><span class="op">)</span> 
<span class="co">#&gt; Loading required package: ape</span>
<span class="fu"><a href="https://rdrr.io/r/base/cat.html">cat</a></span><span class="op">(</span><span class="st">"6 22"</span>,
  <span class="st">"seq_1    --TTACAAATTGACTTATTATA"</span>,
  <span class="st">"seq_2    GATTACAAATTGACTTATTATA"</span>,
  <span class="st">"seq_3    GATTACAAATTGACTTATTATA"</span>,
  <span class="st">"seq_5    GATTACAAATTGACTTATTATA"</span>,
  <span class="st">"seq_8    GATTACAAATTGACTTATTATA"</span>,
  <span class="st">"seq_10   ---TACAAATTGAATTATTATA"</span>,
  file <span class="op">=</span> <span class="st">"matk.phy"</span>, sep <span class="op">=</span> <span class="st">"\n"</span><span class="op">)</span>

  <span class="fu"><a href="https://rdrr.io/r/base/cat.html">cat</a></span><span class="op">(</span><span class="st">"5 15"</span>,
  <span class="st">"seq_1     GATTACAAATTGACT"</span>,
  <span class="st">"seq_3     GATTACAAATTGACT"</span>,
  <span class="st">"seq_4     GATTACAAATTGACT"</span>,
  <span class="st">"seq_5     GATTACAAATTGACT"</span>,
  <span class="st">"seq_8     GATTACAAATTGACT"</span>,
  file <span class="op">=</span> <span class="st">"rbcla.phy"</span>, sep <span class="op">=</span> <span class="st">"\n"</span><span class="op">)</span>

  <span class="fu"><a href="https://rdrr.io/r/base/cat.html">cat</a></span><span class="op">(</span><span class="st">"5 50"</span>,
  <span class="st">"seq_2          GTCTTATAAGAAAGAATAAGAAAG--AAATACAAA-------AAAAAAGA"</span>,
  <span class="st">"seq_3          GTCTTATAAGAAAGAAATAGAAAAGTAAAAAAAAA-------AAAAAAAG"</span>,
  <span class="st">"seq_5          GACATAAGACATAAAATAGAATACTCAATCAGAAACCAACCCATAAAAAC"</span>,
  <span class="st">"seq_8          ATTCCAAAATAAAATACAAAAAGAAAAAACTAGAAAGTTTTTTTTCTTTG"</span>,
  <span class="st">"seq_9          ATTCTTTGTTCTTTTTTTTCTTTAATCTTTAAATAAACCTTTTTTTTTTA"</span>,
  file <span class="op">=</span> <span class="st">"trn1.phy"</span>, sep <span class="op">=</span> <span class="st">"\n"</span><span class="op">)</span>

<span class="fu"><a href="../reference/supermat.html">supermat</a></span><span class="op">(</span>infiles <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"matk.phy"</span>, <span class="st">"rbcla.phy"</span>, <span class="st">"trn1.phy"</span><span class="op">)</span><span class="op">)</span>
<span class="co">#&gt; Supermatrix "supermat.out.phy" and RAxML partition file "gene_partition.txt" have been saved to: </span>
<span class="co">#&gt; /Users/jinlong/Documents/github/phylotools/vignettes</span></code></pre></div>
</div>
<div id="generate-supermatrix-with-fasta-files" class="section level2">
<h2 class="hasAnchor">
<a href="#generate-supermatrix-with-fasta-files" class="anchor"></a>Generate supermatrix with fasta files</h2>
<p>To generate a supermatrix using a matched fasta file, the command is changed to the corresponding fasta file name.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R">
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/helixcn/phylotools">phylotools</a></span><span class="op">)</span>

<span class="fu"><a href="https://rdrr.io/r/base/cat.html">cat</a></span><span class="op">(</span>
  <span class="st">"&gt;seq_1"</span>,  <span class="st">"--TTACAAATTGACTTATTATA"</span>,
  <span class="st">"&gt;seq_2"</span>,  <span class="st">"GATTACAAATTGACTTATTATA"</span>,
  <span class="st">"&gt;seq_3"</span>,  <span class="st">"GATTACAAATTGACTTATTATA"</span>,
  <span class="st">"&gt;seq_5"</span>,  <span class="st">"GATTACAAATTGACTTATTATA"</span>,
  <span class="st">"&gt;seq_8"</span>,  <span class="st">"GATTACAAATTGACTTATTATA"</span>,
  <span class="st">"&gt;seq_10"</span>, <span class="st">"---TACAAATTGAATTATTATA"</span>,
  file <span class="op">=</span> <span class="st">"matk.fasta"</span>, sep <span class="op">=</span> <span class="st">"\n"</span><span class="op">)</span>

<span class="fu"><a href="https://rdrr.io/r/base/cat.html">cat</a></span><span class="op">(</span>
  <span class="st">"&gt;seq_1"</span>, <span class="st">"GATTACAAATTGACT"</span>,
  <span class="st">"&gt;seq_3"</span>, <span class="st">"GATTACAAATTGACT"</span>,
  <span class="st">"&gt;seq_4"</span>, <span class="st">"GATTACAAATTGACT"</span>,
  <span class="st">"&gt;seq_5"</span>, <span class="st">"GATTACAAATTGACT"</span>,
  <span class="st">"&gt;seq_8"</span>, <span class="st">"GATTACAAATTGACT"</span>,
  file <span class="op">=</span> <span class="st">"rbcla.fasta"</span>, sep <span class="op">=</span> <span class="st">"\n"</span><span class="op">)</span>

<span class="fu"><a href="https://rdrr.io/r/base/cat.html">cat</a></span><span class="op">(</span>
  <span class="st">"&gt;seq_2"</span>, <span class="st">"GTCTTATAAGAAAGAATAAGAAAG--AAATACAAA-------AAAAAAGA"</span>,
  <span class="st">"&gt;seq_3"</span>, <span class="st">"GTCTTATAAGAAAGAAATAGAAAAGTAAAAAAAAA-------AAAAAAAG"</span>,
  <span class="st">"&gt;seq_5"</span>, <span class="st">"GACATAAGACATAAAATAGAATACTCAATCAGAAACCAACCCATAAAAAC"</span>,
  <span class="st">"&gt;seq_8"</span>, <span class="st">"ATTCCAAAATAAAATACAAAAAGAAAAAACTAGAAAGTTTTTTTTCTTTG"</span>,
  <span class="st">"&gt;seq_9"</span>, <span class="st">"ATTCTTTGTTCTTTTTTTTCTTTAATCTTTAAATAAACCTTTTTTTTTTA"</span>,
  file <span class="op">=</span> <span class="st">"trn1.fasta"</span>, sep <span class="op">=</span> <span class="st">"\n"</span><span class="op">)</span>

<span class="fu"><a href="../reference/supermat.html">supermat</a></span><span class="op">(</span>infiles <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"matk.fasta"</span>, <span class="st">"rbcla.fasta"</span>, <span class="st">"trn1.fasta"</span><span class="op">)</span><span class="op">)</span>
<span class="co">#&gt; Supermatrix "supermat.out.phy" and RAxML partition file "gene_partition.txt" have been saved to: </span>
<span class="co">#&gt; /Users/jinlong/Documents/github/phylotools/vignettes</span></code></pre></div>
</div>
<div id="citation" class="section level2">
<h2 class="hasAnchor">
<a href="#citation" class="anchor"></a>Citation</h2>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R">
<span class="fu"><a href="https://rdrr.io/r/utils/citation.html">citation</a></span><span class="op">(</span><span class="st">"phylotools"</span><span class="op">)</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; To cite package 'phylotools' in publications use:</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;   Jinlong Zhang (2021). phylotools: Phylogenetic Tools for</span>
<span class="co">#&gt;   Eco-Phylogenetics. R package version 0.2.4.</span>
<span class="co">#&gt;   https://github.com/helixcn/phylotools</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; A BibTeX entry for LaTeX users is</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;   @Manual{,</span>
<span class="co">#&gt;     title = {phylotools: Phylogenetic Tools for Eco-Phylogenetics},</span>
<span class="co">#&gt;     author = {Jinlong Zhang},</span>
<span class="co">#&gt;     year = {2021},</span>
<span class="co">#&gt;     note = {R package version 0.2.4},</span>
<span class="co">#&gt;     url = {https://github.com/helixcn/phylotools},</span>
<span class="co">#&gt;   }</span></code></pre></div>
</div>
<div id="further-reading" class="section level2">
<h2 class="hasAnchor">
<a href="#further-reading" class="anchor"></a>Further reading</h2>
<ul>
<li>Kress, W. J., Erickson, D. L., Jones, F. A., Swenson, N. G., Perez, R., Sanjur, O., &amp; Bermingham, E. (2009). Plant DNA barcodes and a community phylogeny of a tropical forest dynamics plot in Panama. Proceedings of the National Academy of Sciences, 106(44), 18621-18626.</li>
<li>Pei, N., Zhang, J., Mi, X., &amp; Ge, X. (2011). Plant DNA barcodes promote the development of phylogenetic community ecology. Biodiversity Science, 19(3), 284-294.</li>
<li>Roquet, C., Thuiller, W., &amp; Lavergne, S. (2013). Building megaphylogenies for macroecology: taking up the challenge. Ecography, 36(1), 13-26.</li>
<li><a href="https://cme.h-its.org/exelixis/web/software/raxml/index.html" class="uri">https://cme.h-its.org/exelixis/web/software/raxml/index.html</a></li>
</ul>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Jinlong Zhang.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
